specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,940 @@
+        // Gets or sets the sequence number of this notification data.
+        // When multiple ToastProgressData objects are received, the system displays the data with the greatest non-zero number. 
+        UInt32 SequenceNumber;
+
+        // Gets/Sets the value for the title. Binds to {progressTitle} in progress xml tag.
+        String Title;
+
+        // Gets/Sets the Value for the numerical Progress percentile: a number between 0 and 1. Binds to {progressValue} in progress xml tag.
+        Double Value;
+
+        // Gets/Sets the Value for the Progress String. Binds to {progressValueString} in progress xml tag
+        String ValueStringOverride;
+
+        // Gets/Sets the Value for the Status. Binds to {progressStatus} in progress xml tag
+        String Status;
+    };
+
+    [experimental]
+    [flags]
+    // The Toast User Setting or Toast Group Policy Setting
+    enum ToastNotificationSetting
+    {
+        Enabled, // Toast is not blocked by settings or group policy
+        DisabledForApplication, // Toast is blocked by a user defined App Setting
+        DisabledForUser, // Toast is blocked by a user defined Global Setting
+        DisabledByGroupPolicy, // Toast is blocked by Group Policy
+        DisabledByManifest, // Toast is blocked by a setting in the manifest. Only for packaged applications.
+    };
+
+    [experimental]
+    [flags]
+    // Some basic predefined Toast Payload Templates
+    enum ToastTemplateType
+    {
+        BasicToast,
+        ToastWithTwoButtons,
+        ToastWithThreeButtons,
+        ToastWithProgressBar,
+        ToastWithAvatarAndProgressBar,
+        ToastWithAvatar,
+        ToastWithAvatarAndTextBox,
+        ToastWithAvatarAndDropdown,
+        ToastWithDropdown,
+        ToastWithReminder,
+        ToastWithAlarm,
+        ToastWithHeroImage,
+        ToastWithInlineImage,
+    };
+
+    [experimental]
+    [flags]
+    // The Result for a Toast Progress related operation
+    enum ToastProgressResult
+    {
+        Succeeded, // The progress operation succeeded
+        Failed, // The progress operation failed
+        NotificationNotFound, // The progress operation failed to find a toast to process updates
+    };
+
+    [experimental]
+    [flags]
+    enum ToastPriority
+    {
+        Default, // The notification should have default behavior in terms of delivery and display priority during connected standby mode.
+        High, // The notification should be treated as high priority. For desktop PCs, this means during connected standby mode the incoming notification can turn on the screen for Surface-like devices if it doesn't have a closed lid detected.
+    };
+
+    [experimental]
+    // Represent a toast Notification construct
+    runtimeclass ToastNotification
+    {
+        // Initialize a new Toast using an XML Payload.
+        ToastNotification(Windows.Data.Xml.Dom.XmlDocument payload);
+
+        // Unique identifier used to replace a notification within a group.
```

Comment:
Technically the group isn't needed. With a tag you can still do things like remove toast.

Reply:
True. you can get away with replacement of just a tag. But you can also have a tag within a group.